tecken/base/symboldownloader.py

```diff
@@ -97,21 +98,26 @@ def invalidate_cache(self, symbol, debugid, filename):
             prefix = source.prefix
             assert prefix
             file_url = "{}/{}".format(
-                source.base_url, self.make_key(prefix, symbol, debugid, filename)
+                source.base_url, self.make_url_path(prefix, symbol, debugid, filename)
             )
             check_url_head.invalidate(file_url)
 
     @staticmethod
-    def make_key(prefix, symbol, debugid, filename):
-        return "{}/{}/{}/{}".format(
-            prefix,
-            symbol,
-            # The are some legacy use case where the debug ID might
-            # not already be uppercased. If so, we override it.
-            # Every debug ID is always in uppercase.
-            debugid.upper(),
-            filename,
-        )
+    def make_url_path(prefix, symbol, debugid, filename):
+        """Generates a url quoted path which works with HTTP requests against AWS S3
+        buckets
+
+        :arg prefix:
+        :arg symbol:
+        :arg debugid:
+        :arg filename:
+
+        :returns: url quoted relative path to be joined with a base url
+
+        """
+        # The are some legacy use case where the debug ID might not already be
+        # uppercased. If so, we override it. Every debug ID is always in uppercase.
+        return quote(f"{prefix}/{symbol}/{debugid.upper()}/{filename}")
```

Comment:
This is weird, but it'll get rewritten when we redo the storage layer api. Having the docstring comment will help me remember to url quote paths when I do an HTTP storage backend.